docs/faq.rst

```diff
@@ -72,6 +72,54 @@ Additional reading:
 
   * JAX_sharp_bits_
 
+
+How can I control on which device the computation is executed?
+--------------------------------------------------------------
+
+We describe first the principles of data and computation placement
+in JAX.
+
+In JAX the computation follows the data placement. JAX arrays
+have two placement properties: the device where the data resides,
+and whether it is **committed** to the device or not (we sometimes
+say that the data is *sticky* to the device).
+
+By default, JAX arrays are placed uncommitted on the default device
+(``jax.devices()[0]``).
+
+>>> from jax import numpy as jnp
+>>> print(jnp.ones(3).device_buffer.device())
+gpu:0
+
+Computations involving uncommitted data are performed on the default
+device and the results are uncommitted on the default device.
+
+Data can also be placed explicitly on a device using :func:`jax.device_put`,
+in which case if becomes **committed** to the device:
+
+>>> from jax import device_put
+>>> print(device_put(1, jax.devices()[2]).device_buffer.device())
+gpu:2
+
+Computations involving some committed inputs, will happen on the
+committed device, and the result will be committed on the
+same device. It is an error to invoke an operation on
+arguments that are committed to more than one device.
+
+JAX also have a ``device`` parameter for :func:`jax.jit`.
+Jitted functions without a ``device`` parameter behave as any other
+primitive operation (will follow the data and will error
+if invoked on data committed on more than one device).
+
+Jitted computations that have a ``device`` parameter behave as
```

Comment:
As we discussed, I think this is not quite right, at least on master, because of how `jit` will ignore inner `device_put`s. That is, it's not equivalent to just `device_put` on the input arguments.

I wonder if we should actually cut this paragraph and the the first sentence of the preceding one describing `jit`'s `device` parameter, since we expect we might get rid of it.

Reply:
I did cut this and added a note that as of April 2020 there is device parameter that affects this but it is experimental.